src/header.iced

```diff
@@ -0,0 +1,98 @@
+{prng} = require('crypto')
+{createHash} = require('crypto')
+msgpack = require('msgpack-lite')
+crypto = require('keybase-nacl')
+nonce = require('./nonce.iced')
+util = require('./util.iced')
+
+encryption_mode = 0
+attached_sign_mode = 1
+detached_sign_mode = 2
+current_major = 1
+current_minor = 0
+crypto_onetimeauth_BYTES = 32
+crypto_secretkey_BYTES = 32
+
+compute_mac_key = (encryptor, header_hash, pubkey) ->
+  zero_bytes = Buffer.alloc(crypto_onetimeauth_BYTES)
+  mac_box = encryptor.encrypt({plaintext : zero_bytes, nonce : nonce.nonceForMACKeyBox(header_hash), pubkey})
+  return mac_box.slice(-crypto_onetimeauth_BYTES)
+
+exports.generate_encryption_header_packet = (encryptor, recipients) ->
+  header_list = []
+  header_list.push('saltpack')
+  header_list.push([current_major, current_minor])
+  header_list.push(encryption_mode)
+
+  payload_encryptor = crypto.alloc({force_js : true})
+  payload_key = prng(crypto_secretkey_BYTES)
+  payload_encryptor.secretKey = payload_key
+  ephemeral_encryptor = crypto.alloc({force_js : true})
+  ephemeral_encryptor.genBoxPair()
+  header_list.push(ephemeral_encryptor.publicKey)
+
+  sender_sbox = payload_encryptor.secretbox({plaintext : encryptor.publicKey, nonce : nonce.nonceForSenderKeySecretBox()})
+  header_list.push(sender_sbox)
+
+  recipients_list = []
+  for rec_pubkey in recipients
+    rec_pair = []
+    rec_pair.push(rec_pubkey)
```

Comment:
Let's add a `TODO` in here for dealing with anonymous recipients. Maybe that would end up being a third argument, of the same form as `recipients`?